### PR TITLE
STEP SIXTH: 1. pageable added 2. swagger added 3. code style improved

### PR DESCRIPTION
**Pull request by MateuszSeler**

### CODE REVIEW

File: src/main/java/app/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -17,7 +17,6 @@
 
 @ControllerAdvice
 public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
-
```
Comment (slade13) [inline src/main/java/app/exception/CustomGlobalExceptionHandler.java]: Unnecessary blank line removed. While this is not a critical issue, maintaining consistent formatting can improve code readability.

File: src/main/java/app/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -33,8 +32,6 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
                 .map(this::getErrorMessage)
                 .toList();
         errors.put("errors", listOfErrors);
-
```
Comment (slade13) [inline src/main/java/app/exception/CustomGlobalExceptionHandler.java]: Unnecessary blank lines removed. Consistent use of blank lines can help improve code readability by separating logical sections of code.

**Review by slade13 — APPROVED** [review]
Good job on implementing those 3 features, next time please split each feature into separate pull requests. For each task on Mate platform please provide the link to the PR not to the branch.

File: src/main/java/app/model/User.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Data
+@SQLDelete(sql = "UPDATE books SET is_Deleted = true WHERE id = ?")
```
Comment (George-Khomiak) [inline src/main/java/app/model/User.java]: Update **users** not books

File: src/main/java/app/repository/user/UserRepository.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package app.repository.user;
+
+import app.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
```
Comment (George-Khomiak) [inline src/main/java/app/repository/user/UserRepository.java:7]: Nice

File: src/main/java/app/security/AuthenticationService.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package app.security;
+
+import app.dto.user.UserLoginRequestDto;
+
+public interface AuthenticationService {
+    boolean authenticate(UserLoginRequestDto userLoginRequestDto);
+
```
Comment (George-Khomiak) [inline src/main/java/app/security/AuthenticationService.java]: empty line

File: src/main/java/app/mapper/UserMapper.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import app.dto.user.UserResponseDto;
+import app.model.User;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface UserMapper {
+    UserResponseDto toDto(User book);
+
+    User toModel(UserRegistrationRequestDto registrationRequestDto);
+
```
Comment (George-Khomiak) [inline src/main/java/app/mapper/UserMapper.java]: empty line

File: src/main/java/app/service/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    public UserResponseDto register(UserRegistrationRequestDto userRegistrationRequestDto) {
+        if (userRepository.findByEmail(userRegistrationRequestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("User already exists");
+        }
+        if (!userRegistrationRequestDto.getRepeatPassword()
+                .equals(userRegistrationRequestDto.getPassword())) {
+            throw new RegistrationException("Passwords do not match");
+        }
+
+        User user = new User();
```
Comment (George-Khomiak) [inline src/main/java/app/service/UserServiceImpl.java]: Can we use simply userMapper here?

**Review by George-Khomiak — CHANGES_REQUESTED** [review]
Minor comments

File: src/main/java/app/model/User.java (hunk 3 above)
Comment (MateuszSeler) [reply src/main/java/app/model/User.java]: Stupid mistake... Thx!

File: src/main/java/app/service/UserServiceImpl.java (hunk 7 above)
Comment (MateuszSeler) [reply src/main/java/app/service/UserServiceImpl.java]: I simplified the code I as you asked for. I was following the instructions. Apparently, such solution will be used in next lessons.

**Review by George-Khomiak — APPROVED** [review]

**Pull request merged**